meson.build

```diff
@@ -75,44 +75,62 @@ configinc = include_directories('.')
 cc = meson.get_compiler('c')
 dl_dep = cc.find_library('dl', required : false)
 
-WITH_DRM = not get_option('disable_drm') and (host_machine.system() != 'windows')
-libdrm_dep = dependency('libdrm', version : '>= 2.4.60', required : (host_machine.system() != 'windows'))
+backends = ''
+
+WITH_DRM = false
+if not get_option('with_drm').disabled()
+  libdrm_dep = dependency('libdrm', version : '>= 2.4.60', required : (host_machine.system() != 'windows'))
+  WITH_DRM = libdrm_dep.found()
+  if WITH_DRM
+    backends += ' drm'
+  endif
+endif
 
 WITH_X11 = false
-if get_option('with_x11') != 'no'
-  x11_dep = dependency('x11', required : get_option('with_x11') == 'yes')
-  xext_dep = dependency('xext', required : get_option('with_x11') == 'yes')
-  xfixes_dep = dependency('xfixes', required : get_option('with_x11') == 'yes')
-
-  WITH_X11 = (x11_dep.found() and xext_dep.found() and xfixes_dep.found())
-
-  x11_xcb_dep = dependency('x11-xcb', required : get_option('with_x11') == 'yes')
-  xcb_dep = dependency('xcb', required : get_option('with_x11') == 'yes')
-  xcb_dri3_dep = dependency('xcb-dri3', required : get_option('with_x11') == 'yes')
-  WITH_X11 = (WITH_X11 and x11_xcb_dep.found() and xcb_dep.found() and xcb_dri3_dep.found())
+if not get_option('with_x11').disabled()
+  x11_deps = [
+    dependency('x11', required : get_option('with_x11')),
+    dependency('xext', required : get_option('with_x11')),
+    dependency('xfixes', required : get_option('with_x11')),
+    dependency('x11-xcb', required : get_option('with_x11')),
+    dependency('xcb', required : get_option('with_x11')),
+    dependency('xcb-dri3', required : get_option('with_x11')),
+  ]
+  foreach d : x11_deps
+    WITH_X11 = WITH_X11 and d.found()
```

Comment:
WITH_X11 is initially set to false, I think this will always be false then.

Reply:
Thanks fixed :+1: